File: gocode/00xx/0001.go
package gocode

func twoSum(nums []int, target int) []int {
	wanted := make(map[int]int)
	println(wanted[1000])
	for i, v := range nums {
		w, ok := wanted[v]
		if ok {
			return []int{w, i}
		}
		wanted[target-v] = i
	}
	return nil
}

File: pycode/0001.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        want = {}
        for i, v in enumerate(nums):
            if v not in want:
                want[target-v] = i
            else:
                return [want[v], i]
        return []

    def twoSum2(self, nums: list[int], target: int) -> list[int]:
        want: dict[int, int] = {}
        for i, v in enumerate(nums):
            if (lastI := want.get(v)) is not None:
                return [lastI, i]
            want[target-v] = i
        return []

if __name__ == '__main__':
    s = Solution()
    nums = [3, 2, 5, 9, -4, 0, 7]
    target = 1
    print(s.twoSum(nums, target))